wakuv2/waku.go

```diff
@@ -105,21 +106,21 @@ type ErrorSendingEnvelope struct {
 	SentEnvelope SentEnvelope
 }
 
-type ITelemetryClient interface {
+type IMetricsHandler interface {
```

Comment:
While I like the simplicity of this solution (we don't change much code, just change where we publish the metrics now), I think it would be nice to have some common approach for metrics in `status-go`.

There can be many entities that would like to expose it's metrics, e.g. right now it's already:
- `go-waku` (or `nwaku`)
- `libp2p`
- `geth` (will be removed, but theoretically it's still here)

And then we could say that each _service_ in status-go can provide it's own metrics. And the ones you've added here, would be one of such.

What do you think?